spyder/plugins/pylint/widgets/pylintgui.py

```diff
@@ -44,6 +46,31 @@
     import gettext
     _ = gettext.gettext
 
+
+def _find_pylintrc_path(path):
+    os.chdir(path)
+    return pylint.config.find_pylintrc()
+
+
+def get_pylintrc_path(search_paths=None, home_path=None):
+    """Get the path to the highest pylintrc file on a set of search paths."""
+    current_cwd = os.getcwd()
+    pylintrc_path = None
+    if home_path is None:
+        home_path = osp.expanduser("~")

```

Comment:
I think we have a more robust func to do this on https://github.com/spyder-ide/spyder/blob/master/spyder/config/base.py#L153

Reply:
There are two reasons why we don't use this here:

* The path here needs to match the home path used by pylint, which uses `os.path.expanduser("~")`, otherwise the behavior will not be correct, and it raises a RuntimeError in case of home not being found rather than returning None, which is the desired behavior.

* At least under modern Python on Spyder's supported platforms (Windows and *nix), it appears doubtful that `get_home_dir()` is more "robust" in any case. It uses `os.path.expandvars("~")` under the hood, the only differences being that it performs an unnecessary decode from bytes when it is in fact already a string (i.e. unicode) on modern Python, and as a fallback haphazardly tries to get and decode the path from a few env vars, a more carefully-tailored, platform-specific superset of which are already queried by any supported version of [os.path.expanduser](https://docs.python.org/3/library/os.path.html#os.path.expanduser), plus a few strange and useless assignments to the `path` variable that is never used or returned.

This function was originally written by Pierre sometime over [11 years ago before](https://github.com/spyder-ide/spyder/blame/218473f89d9fcd0a48aace045cde4b5e99ff8080/spyderlib/userconfig.py#L55) by Pierre and added in the first commit to this repo from the even earlier Pydee, with only minor modifications as recently 8 years ago, so it is no surprise that it is no longer necessary or a best practice, if it ever was.